src/bsoncxx/types.hpp

```diff
@@ -599,6 +600,42 @@ BSONCXX_INLINE bool operator==(const b_int64& lhs, const b_int64& rhs) {
 }
 
 ///
+/// A BSON Decimal128 value.
+///
+struct BSONCXX_API b_decimal128 {
+    static constexpr auto type_id = type::k_decimal128;
+
+    bsoncxx::decimal128 value;
+
+    ///
+    /// Constructor for b_decimal128.
+    ///
+    /// @param value
+    ///   The value to wrap.
+    ///
+    template <typename T,
+              typename std::enable_if<
+                  !std::is_same<b_decimal128, typename std::decay<T>::type>::value, int>::type = 0>
+    BSONCXX_INLINE explicit b_decimal128(T&& t)
+        : value(std::forward<T>(t)) {
+    }
+
+    ///
+    /// Conversion operator giving a b_decimal128 as a std::string
+    ///
+    std::string to_string();
```

Comment:
Thanks.  That was left over from an earlier design before I decided we needed a standalone bsoncxx::decimal128 class.